x/auth/signing/amino/amino.go

```diff
@@ -0,0 +1,47 @@
+package amino
+
+import (
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	types "github.com/cosmos/cosmos-sdk/types/tx"
+	"github.com/cosmos/cosmos-sdk/x/auth/ante"
+	"github.com/cosmos/cosmos-sdk/x/auth/signing"
+	authtypes "github.com/cosmos/cosmos-sdk/x/auth/types"
+)
+
+// LegacyAminoJSONHandler is a SignModeHandler that handles SIGN_MODE_LEGACY_AMINO_JSON
+type LegacyAminoJSONHandler struct{}
+
+var _ signing.SignModeHandler = LegacyAminoJSONHandler{}
+
+// DefaultMode implements SignModeHandler.DefaultMode
+func (h LegacyAminoJSONHandler) DefaultMode() types.SignMode {
+	return types.SignMode_SIGN_MODE_LEGACY_AMINO_JSON
+}
+
+// Modes implements SignModeHandler.Modes
+func (LegacyAminoJSONHandler) Modes() []types.SignMode {
+	return []types.SignMode{types.SignMode_SIGN_MODE_LEGACY_AMINO_JSON}
+}
+
+// DefaultMode implements SignModeHandler.GetSignBytes
+func (LegacyAminoJSONHandler) GetSignBytes(mode types.SignMode, data signing.SignerData, tx sdk.Tx) ([]byte, error) {
+	if mode != types.SignMode_SIGN_MODE_LEGACY_AMINO_JSON {
```

Comment:
What's the point of passing a mode here? You're already working on a `LegacyAminoJSONHandler`.